extensions/replication/tasks/MultipleBackendTask.js

```diff
@@ -578,6 +702,93 @@ class MultipleBackendTask extends ReplicateObject {
             });
             log.debug('putting data', { entry: destEntry.getLogInfo() });
         }
+        if (sourceEntry.getReplicationIsNFS()) {
+            return this._checkObjectState(sourceEntry, destEntry, log, err => {
+                if (err) {
+                    return doneOnce(err);
+                }
+                return this._sendMultipleBackendPutObject(sourceEntry,
+                    destEntry, part, partObj, size, incomingMsg, log, doneOnce);
+            });
+        }
+        return this._sendMultipleBackendPutObject(sourceEntry, destEntry, part,
+            partObj, size, incomingMsg, log, doneOnce);
+    }
+
+    /**
+     * Get the source object metadata and ensure the latest object MD5 hash is
+     * the same as the entry.
+     * @param {ObjectQueueEntry} sourceEntry - The source object entry
+     * @param {ObjectQueueEntry} destEntry - The destination object entry
+     * @param {Werelogs} log - The logger instance
+     * @param {Function} cb - The callback to call
+     * @return {undefined}
+     */
+    _checkObjectState(sourceEntry, destEntry, log, cb) {
+        return this._getSourceMD(sourceEntry, destEntry, log, (err, res) => {
+            if (err && err.code === 'ObjNotFound' &&
+            !sourceEntry.getIsDeleteMarker()) {
```

Comment:
just a question, if the err code is 'ObjNotFound' and the source entry is a delete marker, why wouldn't this be considered InvalidObjectState?

Reply:
In the case that the source object is deleted we want to replicate the delete marker while skipping any current CRR operations for the previous object (i.e. the one that was just deleted). If it's CRR of a delete marker we are checking the state of, then this is a valid state of the source.